### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ianoti**
This is the pull request of the Amity CP1 project.
The functionality pending implementation is:
`saving to DB and loading from DB`
`docopt frontend interface`

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                return "the room option is invalid"
+            elif not isinstance(rm_name, str):
+                return "the room name is invalid"
+
+    def get_available_space(self, room_type):
+        """ filter the room directory to return available rooms for occupancy """
+        if room_type == "o":
+            office_w_space = [room for room in self.room_directory
+                if (len(room.occupants)< room.capacity) and isinstance(room, Office)]
+            return office_w_space
```
Comment (Stephen-Njoroge) [inline app.py]: The indention of `return office_w_space` is quite suspicious, are you sure it is working and you are getting what you intend?

File: app.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        elif room_type == "l":
+            living_w_space = [room for room in self.room_directory
+                if (len(room.occupants)< room.capacity) and isinstance(room, LivingSpace)]
+            return living_w_space
```
Comment (Stephen-Njoroge) [inline app.py]: if statements end with : and the next lines been controlled by that if statement should be indented under that if.

File: app.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    def get_rooms_w_people(self):
+        """ filter room directory to return rooms with occupants"""
+        rooms_w_people = [room for room in self.room_directory
+            if len(room.occupants)>=1]
+        return rooms_w_people
+
```
Comment (Stephen-Njoroge) [inline app.py:76]: I have seen what you are doing thus deleted the previous comments.

File: app.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                    if len(available_living) >= 1:
+                        allocated_office = random.choice(available_office)
+                        allocated_living = random.choice(available_living)
+                        allocated_living.occupants.append(person)
+                        allocated_office.occupants.append(person)
+                        self.waiting_list.remove(person)
+                    else:
+                        self.living_waiting_list.append(person)
+                        return "Can't allocate Living Space"
+
```
Comment (Stephen-Njoroge) [inline app.py:119]: No need to use `if len(available_office) == 0` and then `elif len(available_office) >= 1:` why not just use'
```
if len(avalilable_office) and len(available_living):
    Do your stuff
print("Can't allocate no Rooms available")

File: app.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python
+import os
+import random
+"""
+    This module implements Amity as a class based interface for the entire
+    room allocation system
+"""
```
Comment (njirap) [inline app.py:7]: maybe not have comments in between imports.. maybe put the comments at the very top..

**Pull request closed without merging**